packages/ui/src/components/navigation/stepper/index.tsx

```diff
@@ -0,0 +1,155 @@
+import { cva } from "class-variance-authority";
+import React from "react";
+import { ReactElement, useCallback } from "react";
+import { Typography } from "../../data-display";
+
+const rootStyles = cva(["cui-flex"], {
+    variants: {
+        layout: {
+            vertical: ["cui-flex-col", "cui-gap-8"],
+            horizontal: ["cui-relative", "cui-flex-row", "cui-gap-2"],
+        },
+    },
+});
+
+const stepStyles = cva(["cui-flex", "cui-items-center", "cui-gap-4"], {
+    variants: {
+        layout: {
+            vertical: [],
+            horizontal: ["cui-flex-col", "cui-w-12"],
+        },
+        clickable: {
+            true: [
+                "hover:cui-underline",
+                "hover:cui-cursor-pointer",
+                "hover:dark:cui-decoration-white",
+            ],
+        },
+    },
+});
+
+const squareStyles = cva(["cui-relative", "cui-h-3", "cui-w-3", "cui-z-10"], {
+    variants: {
+        active: {
+            true: ["cui-bg-orange", "cui-z-10"],
+            false: ["cui-bg-black"],
+        },
+    },
+});
+
+const lineStyles = cva(["cui-absolute"], {
+    variants: {
+        layout: {
+            vertical: [
+                "cui-left-1/2",
+                "cui-top-3",
+                "cui-h-12",
+                "cui-w-[1px]",
+                "-cui-translate-x-[0.5px]",
+                "cui-z-0",
+            ],
+            horizontal: [
+                "cui-top-1/2",
+                "cui-h-[1px]",
+                "cui-w-12",
+                "-cui-translate-y-[0.5px]",
+                "cui-translate-x-3",
+            ],
+        },
+        active: {
+            true: ["cui-bg-orange"],
+            false: ["cui-bg-black"],
+        },
+    },
+});
+
+type StepperLayout = "vertical" | "horizontal";
+
+export interface StepperProps {
+    stepTitles: string[];
+    activeStep: number;
+    mostUpdatedStep: number;
```

Comment:
what does `mostUpdatedStep` means?

Reply:
Maybe `lastStepUpdated`  describes better.

Reply:
The `mostUpdatedStep` is the last completed step, it's used to set the boundary for the navigation; the user can navigate to the already completed steps.

Ye maybe `lastStepCompleted` is a bit more clear.